[13.0][ADD] pos_event_sale

ivantodorovich opened the pull request:
Superseeds: https://github.com/OCA/event/pull/215

![add_event_button](https://user-images.githubusercontent.com/1914185/120641630-34f49400-c44a-11eb-9e27-c31d4a40e937.png)

![event_calendar](https://user-images.githubusercontent.com/1914185/120641638-36be5780-c44a-11eb-9e58-f0da1f8b941a.png)

![ticket_selector](https://user-images.githubusercontent.com/1914185/120641644-38881b00-c44a-11eb-8bbb-e5c9ea04407b.png)

ivantodorovich committed:
[ADD] pos_event_sale : Module to sell events from the Point of Sale.

legalsylvain commented on pos_event_sale/models/pos_config.py:
```diff
@@ -0,0 +1,24 @@
+class PosConfig(models.Model):
+    _inherit = "pos.config"
+
+    iface_event_sale = fields.Boolean("Sell Events")
```
maybe a default to True ? we can consider that if people are installing the module, they want the feature ? (as other pos modules...)

ivantodorovich replied:
I pushed the changes. Although I agree with you, this can have undersired effects during tests if we consider this to be the norm.. (see https://github.com/OCA/pos/pull/642 for example)

This module is only adding to core features, not reducing them.. so it shouldn't be an issue

legalsylvain commented on pos_event_sale/models/pos_order.py:
```diff
@@ -0,0 +1,42 @@
+
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
```
Do we have to handle the refund somewhere in that file ? I guess that refunding a pos.order should cancel the event.registration ? Don't you think ? 

(if not in the scope of your work, maybe add something in the roadmap)

ivantodorovich replied:
Added to roadmap

legalsylvain commented on pos_event_sale/models/pos_order_line.py:
```diff
@@ -0,0 +1,49 @@
+            vals = {
+                "event_ticket_id": ticket_id.id,
+                "event_id": ticket_id.event_id.id,
+                "partner_id": line.order_id.partner_id.id,
```
If I want to buy 2 tickets for two people in the same order, I guess it will not possible because it seems you use the partner_id of the order_id. (or maybe I missed something, I don't use ``event`` module).

ivantodorovich replied:
That's correct. In fact the registration details are not handled currently by this module, as it wasn't in the scope of our project.
In my experience this is not needed most of the times, but I added a comment in the ROADMAP.

legalsylvain reviewed (commented):
very superficial review. a question regarding partner, the rest is minor.

LGTM. Thanks !

ivantodorovich committed:
[IMP] pos_event_sale: Improve usability in touchscreen devices

By default, the longPressDelay is 1000ms which is not intuitive
and makes it pretty much unusable in touchscreen devices.

In this case, it makes no sense to have a delay for selecting dates.
By setting it to 0 we're effectively disabling the pressDelay, and
the user experience is much better.

ivantodorovich committed:
[FIX] pos_event_sale: Customer display

olgamarcocb reviewed (approved):
LGTM 👍

didierdonze reviewed (approved):
Functional review: ok

ivantodorovich commented:
/ocabot merge nobump
